### ISSUE TITLE
[2020/10/30] Identifying Necessary Elements for BERT's Multilinguality

### ISSUE THREAD

**Issue by zaemyung** [zaemyung/Seeking-SOTA#9]
**Identifying Necessary Elements for BERT's Multilinguality**
- EMNLP 2020
- https://arxiv.org/abs/2005.00396
> It has been shown that multilingual BERT (mBERT) yields high quality multilingual representations and enables effective zero-shot transfer. This is surprising given that mBERT does not use any crosslingual signal during training. While recent literature has studied this phenomenon, the reasons for the multilinguality are still somewhat obscure. We aim to identify architectural properties of BERT and linguistic properties of languages that are necessary for BERT to become multilingual. To allow for fast experimentation we propose an efficient setup with small BERT models trained on a mix of synthetic and natural data. Overall, we identify four architectural and two linguistic elements that influence multilinguality. Based on our insights, we experiment with a multilingual pretraining setup that modifies the masking strategy using VecMap, i.e., unsupervised embedding alignment. Experiments on XNLI with three languages indicate that our findings transfer from our small setup to larger scale settings.

**Comment by zaemyung** [comment]
## BERT
![image](https://user-images.githubusercontent.com/3746478/97668864-365c7900-1a83-11eb-9710-097adb164cb4.png)

- Contextualized language model as opposed to autoregressive one.
- Masked language modeling (MLM) + next sentence prediction
- Works well in many downstream NLP tasks

## Multilinguality in NLP
- Helps with task performance in mid-, low-, or even zero-resource languages.
  - E.g. zero-shot translation, natural language inference (NLI), named entity recognition, etc.
- How it arises in models is ongoing research.

### Some findings in related works
- Singh et al. (2019) show that BERT stores language representations in different subspaces and investigate how subword tokenization inﬂuences multilinguality.
- Artetxe et al. (2020) show that neither a shared vocabulary nor joint pretraining is essential for multilinguality.
- K et al. (2020) extensively study reasons for multilinguality (e.g., researching depth, number of parameters, and attention heads). They conclude that depth is essential.
They also investigate language properties and conclude that structural similarity across languages is important, without further deﬁning this term.
- Conneau et al. (2020b) ﬁnd that a shared vocabulary is not required. They ﬁnd that shared parameters in the top layers are required for multilinguality.
Further, they show that different monolingual BERT models exhibit a similar structure and thus conclude that mBERT somehow aligns those isomorphic spaces.
- Conneau et al. (2020a) introduce the term "curse of multilinguality" and show that multilingual model quality degrades with an increased number of languages given a ﬁxed number of parameters.

## This Paper
### Experimental Setting
- First, experiment in a controlled laboratory setting:
  - BERT-small with a single head, only trained with MLM objective
  - Easy-to-Read English Bible, 17k sentence, 2048 vocabs (word-piece tokenized)
  - Construct fake-English corpus as well by shifting the vocab ids.
  ![image](https://user-images.githubusercontent.com/3746478/97670042-b388ed80-1a85-11eb-9748-aa89370e484d.png)
- Then, experiment on crosslingual NLI (XNLI) with BERT-base model on English, German, and Hindi
  - Pretrain on the three language, finetune on English, then do zero-shot on German and Hindi
- Metrics:
  - Word alignment
  - Sentence retrieval
  - Word translation
  - Multilinguality Score = sentence\_retrieval\_0 + word\_translation\_0 + sentence\_retrieval\_8 + word\_translation\_8
- Modifying architectural properties
  - overparam, shift-special (anchor tokens), lang-pos, no-random
- Modifying lingustic properties
  - inv-order, no-parallel
### Results
![image](https://user-images.githubusercontent.com/3746478/97670472-99034400-1a86-11eb-8edc-50c0ee0cac5e.png)
![image](https://user-images.githubusercontent.com/3746478/97670477-9acd0780-1a86-11eb-86cf-9f9753a96c3e.png)
![image](https://user-images.githubusercontent.com/3746478/97670524-b0423180-1a86-11eb-9de7-f439c0c14578.png)

- Similar word order, under-parameterized network capacity, sharing special anchor tokens, and positional embeddings, code-switched words (knn-replacement) are important factors in multilinguality.
